app/src/main/java/org/simple/clinic/widgets/qrcodescanner/QrCodeScannerView.kt

```diff
@@ -30,6 +32,8 @@ class QrCodeScannerView(context: Context, attrs: AttributeSet) : FrameLayout(con
   @Inject
   lateinit var lifecycle: Observable<TheActivityLifecycle>
 
+  private val qrCodeScansValve = PublishSubject.create<Boolean>()
```

Comment:
Since this is a `PublishSubject`, and we're using `withLatestFrom` to pause/resume the stream, the QR code scanning will never work unless we open and close the keyboard at least once. Should this be a `BehaviorSubject` with an initial value of `true`?

Reply:
It was a `BehaviorSubject` 😮 good catch BTW.